.github/workflows/knip.yml

```diff
@@ -0,0 +1,28 @@
+name: Knip
+
+on:
+  pull_request:
+
+permissions: write-all
+
+jobs:
+  run-knip:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.14.1"
```

Comment:
We should use latest LTS

Reply:
good point, I will check with the latest LTS as well

Reply:
Is this finished? Should I merge?

Reply:
@pavlovcik I will test with latest LTS today and let you know asap.

Reply:
I rebased to latest knip v3 support in my fork, fixed the tests and opened a PR upstream https://github.com/Codex-/knip-reporter/pull/15 will try to push this, if this is accepted we will be able to use an official release.

Reply:
My pull request was extended and approved, merged upstream. I will now verify this on my fork and let you know if this works as expected.

Reply:
I spent quite some time on debugging and asked the author why it does not work as expected on latest upstream release https://github.com/Codex-/knip-reporter/issues/9#issuecomment-1853771495 but does work on my fork let's wait for his response.

Reply:
@pavlovcik after a few hours I found the culprit and solved the problem.

knip v3 does not work well with typescript < 5 , does not throw errors but its output json does not contain all needed data. I had to update ubiquibot's dev dependencies to typescript ^5.0.4 and jest + ts-jest to v29 and it works like a charm, annotations in action are visible in my qa at https://github.com/gitcoindev/ubiquibot/pull/17/files

It is a bit late for me today, I will check tomorrow if this did not break anything and will add the typescript and jest upgrade as well to refactor/general